fastify.js

```diff
@@ -718,6 +743,29 @@ function fastify (options) {
     opts.includeMeta = opts.includeHooks ? opts.includeMeta ? supportedHooks.concat(opts.includeMeta) : supportedHooks : opts.includeMeta
     return router.printRoutes(opts)
   }
+
+  function wrapRouting (router, { rewriteUrl, logger }) {
+    let isAsync
+    return function preRouting (req, res) {
+      // only call isAsyncConstraint once
+      if (isAsync === undefined) isAsync = router.isAsyncConstraint()
+      if (!rewriteUrl) {
+        router.routing(req, res, buildAsyncConstraintCallback(isAsync, req, res))
+      } else {
+        const originalUrl = req.url
+        const url = rewriteUrl(req)
+        if (originalUrl !== url) {
+          logger.debug({ originalUrl, url }, 'rewrite url')
+          if (typeof url === 'string') {
+            req.url = url
+          } else {
+            req.destroy(new Error(`Rewrite url for "${req.url}" needs to be of type "string" but received "${typeof url}"`))
```

Comment:
```suggestion
            req.destroy(new Error(`Rewrite url for "${req.url}" needs to be of type "string" but received "${typeof url}"`))
            return
```

Reply:
I would just keep it for now. It retain the old behavior.
Calling handler even when `request` is destroyed.